### ISSUE TITLE
Unresponsive to controller input

### ISSUE THREAD

**Issue by PatrickvL** [Cxbx-Reloaded/Cxbx-Reloaded#1222]
Somewhere in the past few days (from around 2018-05-20) a few titles stopped responding to input.

Test cases:
- [x] Crazy Taxi 3 - High Roller
- [x] Lensflare xdk sample
- [ ] [Sonic Riders](https://github.com/Cxbx-Reloaded/game-compatibility/issues/451#issuecomment-392906305)
- [x] [Star Wars: Knights of the Old Republic](https://github.com/Cxbx-Reloaded/game-compatibility/issues/428#issuecomment-392844117)
- [x] [Star Wars Knights of the Old Republic II](https://github.com/Cxbx-Reloaded/game-compatibility/issues/96#issuecomment-392869872)

**EDIT:** Convert unordered list into task list for status report.

**Comment by PatrickvL** [comment]
This might be caused by https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/commit/ab64d5e96ac5575a7dac89a89cc1fe6929cec57d

**Comment by RadWolfie** [comment]
Source of cause:
https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/blob/2b52b6f91832a26bc74afa7c394f424d2c7af632/src/CxbxKrnl/EmuXapi.cpp#L756

For some reason device info internally with variable named subType from XID_TYPE_INFORMATION could be wrong.
Value obtained from source:
https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/blob/2b52b6f91832a26bc74afa7c394f424d2c7af632/src/CxbxKrnl/EmuXapi.h#L179

It might be possible to obtain the value directly from hardware device or some sort of switch/conversion in the official xdk.

Real sample obtain from kotor:
```
DeviceTable[0]->ucType = 1, ucSubType = 4
DeviceTable[0]->XppType = 0x003DEF8C (XDEVICE_TYPE_GAMEPAD)
----------------------------------------
DeviceTable[1]->ucType = 3, ucSubType = 1
DeviceTable[1]->XppType = 0x003DEFF0 (Unknown device type)
```

Gamepad's SubType for XAPI level is 1, not 4 by the way.

**Comment by RadWolfie** [comment]
Here's [the controller research branch](https://github.com/RadWolfie/Cxbx-Reloaded/tree/xapi-fix-tests) I made to investigate the source of cause with workaround fix.

I personally doubt it will fix other hardware device for xbox anyway as I am not sure which device type binds to specific subtype.

**Comment by jackchentwkh** [comment]
The subtype shall be 0x04 for gamepad. 
In my previous code change I might use wrong value. It could be my fault. I am it for a meeting won’t be available for a while. If using subtype 0x04 fix the problem, that should be it.

**Comment by LukeUsher** [comment]
XID_TYPE_INFORMATION struct layout appears to be wrong. Investigating the correct struct layout

**Comment by jagotu** [comment]
At least the LensFlare demo doesn't seem to even have XInputSetState, so #1194 probably isn't the culprit.

**Comment by x1nixmzeng** [comment]
I changed the layout by packing `XINPUT_FEEDBACK_HEADER` as part of #680.

See: https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/commit/2fb4ea73de37e4c678a48ae59c52c4645e60d097

Which I suspect is the structure changed in one of the XDKs and my change doesn't consider it.

**Comment by PatrickvL** [comment]
In Lensflare, XInput keyboard input *is* detected in `XBController::ListenPoll` (easily proven by putting a breakpoint below `if(bKey & 0x80)`) - it's just that there's no response to the input.

**Comment by RadWolfie** [comment]
Currently in develop branch, these titles are responding input. (Might be due to render screen wasn't update properly.)

* Crazy Taxi 3 - High Roller
* Lensflare xdk sample
* Star Wars: Knights of the Old Republic
* Star Wars Knights of the Old Republic II

However, untested in master branch.

**Comment by PatrickvL** [comment]
What about Sonic Riders?

**Comment by PatrickvL** [comment]
Last fix is in #1380

**Comment by RadWolfie** [comment]
In my opinion, the *real* last fix is #1333. Since Star Wars: KotOR was reported as fix. So basically other titles should work fine as well.
